From 551b95552aa919b2784aeff1a4f902098d1f5caa Mon Sep 17 00:00:00 2001
From: Hardik T Shah <hardik.t.shah@intel.com>
Date: Thu, 10 Mar 2016 22:25:41 +0530
Subject: [PATCH 3865/4301] SoundWire: Intel: Add support to configure mater
 ports.

commit ac5d7522c22f29daf842b1b6d3d340b7c59e9cdb from
git://git@github.com:01org/linux-intel-4.9.git

This method registers the port configuration method
with bus driver. Since SoundWire master controller
register map is not defined by MIPI spec, master controller
provides methods to be used by bus driver for master
controller port setup. Port setup is done by the
bus driver as part of stream setup between
SoundWire devices (Master(s) and Slave(s))

Change-Id: I6592102db8c4625822a795918059fd255fbb4780
Signed-off-by: Hardik T Shah <hardik.t.shah@intel.com>
---
 drivers/sdw/sdw_cnl.c |   42 ++++++++++++++++++++++++++++++++++++++++++
 1 files changed, 42 insertions(+), 0 deletions(-)

diff --git a/drivers/sdw/sdw_cnl.c b/drivers/sdw/sdw_cnl.c
index 0c2a34f..7405c87 100644
--- a/drivers/sdw/sdw_cnl.c
+++ b/drivers/sdw/sdw_cnl.c
@@ -745,6 +745,35 @@ static int cnl_sdw_set_clock_freq(struct sdw_master *mstr,
 	return 0;
 }
 
+static int cnl_sdw_set_port_params(struct sdw_master *mstr,
+			struct sdw_port_params *params, int bank)
+{
+	return 0;
+}
+
+static int cnl_sdw_set_port_transport_params(struct sdw_master *mstr,
+			struct sdw_transport_params *params, int bank)
+{
+	return 0;
+}
+
+static int cnl_sdw_port_activate_ch(struct sdw_master *mstr,
+			struct sdw_activate_ch *activate_ch, int bank)
+{
+	return 0;
+}
+
+static int cnl_sdw_port_activate_ch_pre(struct sdw_master *mstr,
+			struct sdw_activate_ch *activate_ch, int bank)
+{
+	return 0;
+}
+static int cnl_sdw_port_activate_ch_post(struct sdw_master *mstr,
+			struct sdw_activate_ch *activate_ch, int bank)
+{
+	return 0;
+}
+
 static int cnl_sdw_probe(struct sdw_master *mstr,
 				const struct sdw_master_id *sdw_id)
 {
@@ -808,6 +837,18 @@ static struct sdw_master_ops cnl_sdw_master_ops  = {
 	.set_frame_shape = NULL,
 };
 
+static struct sdw_master_port_ops cnl_sdw_master_port_ops = {
+	.dpn_set_port_params = cnl_sdw_set_port_params,
+	.dpn_set_port_transport_params = cnl_sdw_set_port_transport_params,
+	.dpn_port_activate_ch = cnl_sdw_port_activate_ch,
+	.dpn_port_activate_ch_pre = cnl_sdw_port_activate_ch_pre,
+	.dpn_port_activate_ch_post = cnl_sdw_port_activate_ch_post,
+	.dpn_port_prepare_ch = NULL,
+	.dpn_port_prepare_ch_pre = NULL,
+	.dpn_port_prepare_ch_post = NULL,
+
+};
+
 static struct sdw_mstr_driver cnl_sdw_mstr_driver = {
 	.driver_type = SDW_DRIVER_TYPE_MASTER,
 	.driver = {
@@ -816,6 +857,7 @@ static struct sdw_mstr_driver cnl_sdw_mstr_driver = {
 	.probe          = cnl_sdw_probe,
 	.remove         = cnl_sdw_remove,
 	.mstr_ops	= &cnl_sdw_master_ops,
+	.mstr_port_ops = &cnl_sdw_master_port_ops,
 };
 
 static int __init cnl_sdw_init(void)
-- 
1.7.5.4

